ocs_ci/utility/networking.py

```diff
@@ -0,0 +1,85 @@
+"""
+Module that contains network related functions
+"""
+
+import ipaddress
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.exceptions import (
+    DeploymentPlatformNotSupported,
+    UnavailableResourceException,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def annotate_worker_nodes_with_mon_ip():
+    """
+    Annotate worker nodes with annotation 'network.rook.io/mon-ip: <IPAddress>'
+    """
+    if not config.ENV_DATA["platform"].lower() in constants.BAREMETAL_PLATFORMS:
+        raise DeploymentPlatformNotSupported(
+            "Annotating nodes with mon ip is not implemented for current platform"
+        )
+
+    nodes_obj = OCP(kind="node")
+    nodes = nodes_obj.get().get("items", [])
+    worker_nodes = [
+        node["metadata"]["name"]
+        for node in nodes
+        if constants.WORKER_LABEL in node["metadata"]["labels"]
+    ]
+    if not worker_nodes:
+        raise UnavailableResourceException("No worker node found!")
+    for worker in worker_nodes:
+        network_data = (
+            config.ENV_DATA.get("baremetal", {}).get("servers", {}).get(worker)
+        )
+        annotate_cmd = (
+            f"annotate node {worker} "
+            f"network.rook.io/mon-ip={network_data['private_ip']} --overwrite"
+        )
```

Comment:
Here it might be worth to check if `private_ip` exists in the `network_data` structure and if not, raise `ConfigurationError` with explanation (or cover this in `try`...`except` block and re-raise `KeyError` exception as ` ConfigurationError`. It will require to add `{}` as default value also to the last `get` on the line 40.

Reply:
From the test log it should be clear if private_ip is missing - key error exception will be raised when trying to access it in the f-string.